integration-test/tanstack-start/app/routes/preloadQuery/useSuspenseQuery.tsx

```diff
@@ -0,0 +1,48 @@
+import { QUERY } from "@integration-test/shared/queries";
+import { useSuspenseQuery, type DefaultContext } from "@apollo/client/index.js";
+import "@integration-test/shared/errorLink";
+import { createFileRoute } from "@tanstack/react-router";
+import { Suspense } from "react";
+
+export const Route = createFileRoute("/preloadQuery/useSuspenseQuery")({
+  component: RouteComponent,
+  validateSearch: (search: Record<string, unknown>) => {
+    return {
+      errorIn: search.errorIn as DefaultContext["error"],
+    };
+  },
+  loaderDeps: ({ search: { errorIn } }) => ({ errorIn }),
+  loader: async ({ context: { preloadQuery }, deps: { errorIn } }) => {
+    const queryRef = preloadQuery(QUERY, {
+      context: { delay: 1000, ...(errorIn ? { error: errorIn } : {}) },
+    });
+    return {
+      queryRef,
```

Comment:
I'm curious how `queryRef` plays a role here since its not used in `Child` or `RouteComponent`. Would you mind explaining that one? Is just calling `preloadQuery` enough?

Reply:
Just calling `preloadQuery` won't do anything - the signal that the data should go over the network is the `queryRef` itself.
The `queryRef` will get hydrated on arrival in the browser and start filling the cache - for that part it doesn't matter if you actually use it.
Compare this to `useBackgroundQuery` without actually listening on the `ref` - the network request happens anyways.